pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -292,12 +290,26 @@ func (r *Reconciler) getReceiveAdapter(ctx context.Context, src *v1alpha1.CronJo
 	return nil, apierrors.NewNotFound(schema.GroupResource{}, "")
 }
 
-func (r *Reconciler) createEventType(ctx context.Context, src *v1alpha1.CronJobSource) (*eventingv1alpha1.EventType, error) {
+func (r *Reconciler) reconcileEventType(ctx context.Context, src *v1alpha1.CronJobSource) (*eventingv1alpha1.EventType, error) {
 	current, err := r.getEventType(ctx, src)
 	if err != nil && !apierrors.IsNotFound(err) {
 		logging.FromContext(ctx).Error("Unable to get an existing event type", zap.Error(err))
 		return nil, err
 	}
+
+	// Only create EventTypes for Broker sinks. But if there is an EventType and the src has a non-Broker sink
+	// (possibly because it was updated), then we need to delete it.
```

Comment:
Why is this different from the apiserversource ?

Reply:
The apiserversource creates a list of EventTypes, here is just one. 
I could have done something similar as in the eventing-sources PR, but this was more of a last minute change. Let's leave it as is for now, and then I can refactor it.

Reply:
Thanks for the comment BTW